src/VisualStudio/IntegrationTest/TestUtilities/InProcess/TextViewWindow_InProc.cs

```diff
@@ -269,13 +269,26 @@ public void InvokeQuickInfo()
         }
 
         public string GetQuickInfo()
-            => ExecuteOnActiveView(view =>
+        {
+            return ThreadHelper.JoinableTaskFactory.Run(async () =>
             {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                var view = GetActiveTextView();
                 var broker = GetComponentModelService<IAsyncQuickInfoBroker>();
 
                 var session = broker.GetSession(view);
+
+                using var cts = new CancellationTokenSource(Helper.HangMitigatingTimeout);
+                while (session.State != QuickInfoSessionState.Visible)
```

Comment:
Two options on how to do this:

- I think awaiting `TriggerQuickInfoAsync()` will ensure that the tooltip is active before returning the session.
- Alternatively, if you don't care about the UI being visible, you can call `GetQuickInfoItemsAsync()` to just get the content that would be disabled in the tooltip.

Reply:
> think awaiting `TriggerQuickInfoAsync()` will ensure that the tooltip is active before returning the session

We call `InvokeQuickInfo` before calling `GetQuickInfo`, and I recently updated the former to await `TriggerQuickInfoAsync`.

> Alternatively, if you don't care about the UI being visible, you can call `GetQuickInfoItemsAsync()` to just get the content that would be displayed in the tooltip.

Since this is integration tests, I think we should check the actual content if possible.

Reply:
I'm surprised this necessary, AFAICT we're not doing any fire-and-forget async work. It should all be tracked by the task returned by Trigger.

Reply:
That's exactly what I was thinking, but the log is showing test failures even though the content is clearly appearing.

Reply:
Exceptions and screenshots can be seen in the artifacts for https://dev.azure.com/dnceng/public/_build/results?buildId=1073724&view=results

Reply:
Should we get a dump created at the failure point? It seems like this should be relatively straightforward for @gundermanc or somebody else to debug, right?

Reply:
Sure, if you can point me to some sort of diagnostic info I can take a look opportunistically.

Reply:
We don't have a dump. Even by the time the screenshot is taken for the test failure exception you can see the expected Quick Info content.